x/ccv/consumer/keeper/validators.go

```diff
@@ -61,24 +65,58 @@ func (k Keeper) ApplyCCValidatorChanges(ctx sdk.Context, changes []abci.Validato
 	return ret
 }
 
-// IterateValidators - unimplemented on CCV keeper but perform a no-op in order to pass the slashing module InitGenesis.
-// It is allowed since the condition verifying validator public keys in HandleValidatorSignature (x/slashing/keeper/infractions.go) is removed
-// therefore it isn't required to store any validator public keys to the slashing states during genesis.
-func (k Keeper) IterateValidators(sdk.Context, func(index int64, validator stakingtypes.ValidatorI) (stop bool)) {
+// IterateValidators iterate democracy staking validators when the block height is before the last sovereign height
```

Comment:
All of this is about dealing with how to punish the old valset. We need the old staking keeper for at least the unbonding period